restore MAIN_SHOWDATABASENAMEINHELPPAGESLINK option to show database name in help pages link

bafbes opened the pull request:
# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "Fix", "Close" or "New" section*
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed textswith meaningful informations*


# Fix #[*issue_number Short description*]
[*Long description*]


# Close #[*issue_number Short description*]
[*Long description*]


# New [*Short description*]
[*Long description*]

bafbes committed:
Fix: Database name may not be contained in $db->databasename

eldy commented on htdocs/main.inc.php:
```diff
@@ -1641,6 +1641,10 @@ function top_menu($head, $title = '', $target = '', $disablejs = 0, $disablehead
 				$text = '';
 				$title = $langs->trans($mode == 'wiki' ? 'GoToWikiHelpPage' : 'GoToHelpPage');
 				if ($mode == 'wiki') $title .= ' - '.$langs->trans("PageWiki").' &quot;'.dol_escape_htmltag(strtr($helppage, '_', ' ')).'&quot;'."";
+	            if (!empty($conf->global->MAIN_SHOWDATABASENAMEINHELPPAGESLINK)) {
```
It seems this option is already managed few lines later (line 1658) toshow database name on the popup of version.

bafbes committed:
Update main.inc.php

The database name appears sometimes deeper in the $db object, so I removed the new inclusion and fixed the old.

eldy commented on htdocs/main.inc.php:
```diff
@@ -1653,7 +1653,7 @@ function top_menu($head, $title = '', $target = '', $disablejs = 0, $disablehead
 			if (!empty($conf->global->MAIN_SHOWDATABASENAMEINHELPPAGESLINK)) {
 				$langs->load('admin');
-				$appli .= '<br>'.$langs->trans("Database").': '.$db->database_name;
+                $title .= '<br>'.$langs->trans("Database").': '.(!empty($db->database_name)?$db->database_name:$db->db->database_name);
```
Do you mean you have some cases where $db->db->database_name and $db->database_name is not ?
When on a page, $db->database_name should always be defined i think.

bafbes replied:
I did not record the commit where the database name was only accessible through $db->db->database_name but as I use always this option I have been annoyed not see it due  to this attribute moving. So I opted to solve it this way.

eldy replied:
$db should always be Database handler. If not i means you have some bad code that erase it with an object that is a common object class. In such a case, you should fix bad code instead. I suggest to ait you find the case because this change is for a case that should never happen.

bafbes commented:
In this case this may be closed
